MRFIXIT: ALLOWS FOR THE OVERRIDE OF THE GENERAL VIDEO RESOLUTION SETTINGS INSIDE THE GLIDEN64 SETTINGS IN MUPEN64PLUS.CFG
--- a/src/mupenplus/Config_mupenplus.cpp
+++ b/src/mupenplus/Config_mupenplus.cpp
@@ -35,14 +35,18 @@
 	// Set default values for "Video-General" section, if they are not set yet. Taken from RiceVideo
 	m64p_error res = ConfigSetDefaultBool(g_configVideoGeneral, "Fullscreen", config.video.fullscreen, "Use fullscreen mode if True, or windowed mode if False ");
 	assert(res == M64ERR_SUCCESS);
 	res = ConfigSetDefaultInt(g_configVideoGeneral, "ScreenWidth", config.video.windowedWidth, "Width of output window or fullscreen width");
 	assert(res == M64ERR_SUCCESS);
 	res = ConfigSetDefaultInt(g_configVideoGeneral, "ScreenHeight", config.video.windowedHeight, "Height of output window or fullscreen height");
 	assert(res == M64ERR_SUCCESS);
+	res = ConfigSetDefaultInt(g_configVideoGliden64, "ScreenWidth", config.video.windowedWidth, "Width of output window or fullscreen width");
+	assert(res == M64ERR_SUCCESS);
+	res = ConfigSetDefaultInt(g_configVideoGliden64, "ScreenHeight", config.video.windowedHeight, "Height of output window or fullscreen height");
+	assert(res == M64ERR_SUCCESS);
 	res = ConfigSetDefaultInt(g_configVideoGliden64, "ScreenLeftOffset", config.video.screenLeftOffset, "Position of the left of the screen");
 	assert(res == M64ERR_SUCCESS);
 	res = ConfigSetDefaultBool(g_configVideoGeneral, "VerticalSync", config.video.verticalSync, "If true, activate the SDL_GL_SWAP_CONTROL attribute");
 	assert(res == M64ERR_SUCCESS);
 
 	res = ConfigSetDefaultInt(g_configVideoGliden64, "configVersion", CONFIG_VERSION_CURRENT, "Settings version. Don't touch it.");
 	assert(res == M64ERR_SUCCESS);
@@ -208,10 +208,15 @@
 		}
 	}
 
 	config.video.fullscreen = ConfigGetParamBool(g_configVideoGeneral, "Fullscreen");
-	config.video.windowedWidth = ConfigGetParamInt(g_configVideoGeneral, "ScreenWidth");
-	config.video.windowedHeight = ConfigGetParamInt(g_configVideoGeneral, "ScreenHeight");
+	if (ConfigGetParamInt(g_configVideoGliden64, "ScreenWidth") == 0 || ConfigGetParamInt(g_configVideoGliden64, "ScreenHeight") == 0) {
+		config.video.windowedWidth = ConfigGetParamInt(g_configVideoGeneral, "ScreenWidth");
+		config.video.windowedHeight = ConfigGetParamInt(g_configVideoGeneral, "ScreenHeight");
+	} else {
+		config.video.windowedWidth = ConfigGetParamInt(g_configVideoGliden64, "ScreenWidth");
+		config.video.windowedHeight = ConfigGetParamInt(g_configVideoGliden64, "ScreenHeight");
+	}
 	config.video.screenLeftOffset = ConfigGetParamInt(g_configVideoGliden64, "ScreenLeftOffset");
 	config.video.verticalSync = ConfigGetParamBool(g_configVideoGeneral, "VerticalSync");
 
 	config.video.cropMode = ConfigGetParamInt(g_configVideoGliden64, "CropMode");

